jdaviz/core/template_mixin.py

```diff
@@ -1057,6 +1058,32 @@ def __init__(self, *args, **kwargs):
                                            allowed_type='spatial')
 
 
+class DatasetSpectralSubsetValidMixin(VuetifyTemplate, HubListener):
+    """
+    Adds a traitlet tracking whether self.dataset and self.spectral_subset
+    overlap in the spectral axis.
+
+    Note that if using in another method that is also observing dataset_selected
+    or spectral_subset_selected, that that method could be called before the traitlet
+    is updated.  In that case, call self._check_dataset_spectral_subset_valid() itself
+    directly.  The traitlet can still be used for any warning text in the plugin UI.
+    """
+    spectral_subset_valid = Bool(True).tag(sync=True)
+
+    @observe("dataset_selected", "spectral_subset_selected")
+    def _check_dataset_spectral_subset_valid(self, event={}):
+        # TODO: does this window not account for gaps?  Should we add the warning?
+        # or can this be removed (see note above in _dataset_selected_changed)
+        if self.spectral_subset_selected == "Entire Spectrum":
+            self.spectral_subset_valid = True
+        else:
+            spec = self.dataset.selected_obj
+            spec_min, spec_max = np.nanmin(spec.spectral_axis), np.nanmax(spec.spectral_axis)
+            subset_min, subset_max = self.spectral_subset.selected_min_max(spec)
+            self.spectral_subset_valid = bool(subset_min < spec_max and subset_max > spec_min)
```

Comment:
Why is the range check not inclusive?
```suggestion
            self.spectral_subset_valid = bool(subset_min =< spec_max and subset_max >= spec_min)
```

Reply:
We need to ensure the subset will contain data.  From testing it seemed that isn't the case for exact edge values, but that could just be due to rounding error or might be related to #1892.  Can you point to where in the code the subset logic is guaranteed to be inclusive?  I don't mind changing this if it should act inclusively, but also want to avoid a rabbit hole here and might just need to revisit this during/after the possible upcoming subset refactoring work.

Reply:
I don't know on top of my head if it should be inclusive or not. I just want to make sure this was considered. Thanks for the clarification!